explorer/interpreter/type_checker.cpp

```diff
@@ -1700,31 +1700,31 @@ class TypeChecker::ConstraintTypeBuilder {
     friend class ConstraintTypeBuilder;
 
    private:
-    int num_impls_added = 0;
-    int num_equals_added = 0;
+    int num_impls_constraints_added = 0;
+    int num_equality_constraints_added = 0;
```

Comment:
Maybe `num_equals_constraints_added`, given that we call the representation an `EqualsConstraint`.